documentation/internal/Filters-for-Applications.md

```diff
@@ -0,0 +1,292 @@
+
+##### consolidator
+
+Example: `consolidator="Tony"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `consolidator="Tony","Andrei"`
+
+Options:
+
+- `"username"` - Consolidator username
+
+##### applicant
+
+Example: `applicant="John"`
+Considered in implementation: **No**
+Can have combined values: **Yes** e.g: `applicant="John","Nicole"`
+
+Options:
+
+- `"username"` - Applicant username
+
+##### org
+
+Example: `org="Company A"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `org="Company A","Company B"`
+
+Options:
+
+- `"organisation name"`
+
+##### search
+
+Example: `search="abc 123"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **No**
+
+Options:
+
+- String containing [A-Z], [a-z], [0-9], _space_
+- Uses _starts with_ on search including following columns: "Application name", "Stage", ... (**TODO: continue listing columns**)
+- (See more rules on Formats & Rules)
+
+---
+
+#### Date filters:
+
+##### last-active-date
+
+Example: `last-active-date=2021-01-01`
+Considered in implementation: **Yes** [#228 - Dates in columns]
+Can have combined values: **Yes** e.g: `last-active-date=today,last-week`
+
+Options:
+
+- Pre-defined string: `today` (See more formats on Formats & Rules)
+- Single date: `YYYY-MM-DD`
+- Period (start:end): `YYYY-MM-DD:YYYY-MM-DD`
+
+##### deadline-date
+
+Example: `deadline-date=2021-01-31`
+Considered in implementation: **Yes** [#228 - Dates in columns]
+Can have combined values: **Yes** e.g: `deadline-date=today,2021-01-02`
+
+Options:
+
+- Pre-defined string: `today` (See more formats on Formats & Rules)
+- Single date: `YYYY-MM-DD`
+- Period (start:end): `YYYY-MM-DD:YYYY-MM-DD`
+
+---
+
+#### Integer filters:
+
+##### page
+
+Example: `page=1`
+Considered in implementation: **Yes** [#210 - Pagination]
+Can have combined values: **No**
+
+Options:
+
+- Number between 1 - N (maximum of pages?)
+
+##### per-page
+
+Example: `per-page=20`
+Considered in implementation: **Yes** [#210 - Pagination]
+Can have combined values: **No**
+
+Options:
+
+- Number between 10 - 100
+
+---
+
+### Formats & Rules
+
+---
+
+#### Pre-defined dates
+
+`today`, `yesterday`, `this-week`, `last-week`, `this-month`, `last-month`, `this-quarter`, `last-quarter`,`this-year`, `last-year`, ... (Add other pre-defined date here)
+
+#### Strings
+
+Can't have special caracters.
+Are case insensitive. So `thisTHAT` = `THISthat` = `thisthat`.
+
+#### Labels
+
+When combined labels are used (with separator `,`) the query will be using the OR connector.
+
+#### General
+
+When combined filters are used (with separator `&`) the query will be using the AND connector.
+
+---
+
+### Columns per user-role
+
+---
+
+### Common columns (For all user-roles)
+
+- Last active date
+- Application name
+- Stage
+- Status / Actionable
+
+#### Applicant
+
+- Deadline date
+- Progress
+
+#### Reviewer
+
+- Company
```

Comment:
We actually have no way to tie an application to an organisation currently (applications only have user_id), so I've made an issue for it: [Back-end #179](https://github.com/openmsupply/application-manager-server/issues/179)

Reply:
Also, I think we need to keep getting in the habit of using "Organisation" rather than "Company".

Reply:
Agreed. The filter name is already `org` so all good.